modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.shared/src/main/scala/edu/gemini/seqexec/web/common/LogMessage.scala

```diff
@@ -0,0 +1,45 @@
+package edu.gemini.seqexec.web.common
+
+import java.util.logging.{Level, LogRecord}
+
+import upickle.Js
+
+import scala.collection.breakOut
+
+/** Minimal class to transfer log messages from client to server
+  * We only care about level and message
+  */
+case class LogMessage(level: Level, msg: String)
+
+object LogMessage {
+  def fromLogRecord(record: LogRecord): LogMessage = {
+    // We may want to encode time, but the timezone on the client cannot be
+    // easily recorded. Discard exceptions too
+    LogMessage(record.getLevel, record.getMessage)
+  }
+
+  // Map levels to numbers to reduce space
+  private val Int2LevelMapping: Map[Int, Level] = Map(0 -> Level.OFF,
+    1 -> Level.SEVERE,
+    3 -> Level.WARNING,
+    4 -> Level.INFO,
+    5 -> Level.CONFIG,
+    6 -> Level.FINE,
+    7 -> Level.FINER,
+    8 -> Level.FINEST,
+    9 -> Level.ALL)
```

Comment:
You skipped `2`?  I suppose you could also make a `List` in the order you want, zip with index, and turn it into a `Map`.  Is it worth worrying about saving space?  The text of the message itself is presumably much larger anyway?